core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -679,18 +682,25 @@ class ReplicaManager(val config: KafkaConfig,
     for (partition <- partitionState.keys)
       responseMap.put((partition.topic, partition.partitionId), ErrorMapping.NoError)
 
+    val partitionsToMakeLeaders: mutable.Set[Partition] = mutable.Set()
+
     try {
       // First stop fetchers for all the partitions
       replicaFetcherManager.removeFetcherForPartitions(partitionState.keySet.map(new TopicAndPartition(_)))
-      partitionState.foreach { state =>
+      // Update the partition information to be the leader
+      partitionState.foreach{ case (partition, partitionStateInfo) =>
+        if (partition.makeLeader(controllerId, partitionStateInfo, correlationId))
+          partitionsToMakeLeaders += partition
+        else
+          stateChangeLogger.info(("Broker %d skipped the become-leader state change after marking its partition as leader with correlation id %d from " +
```

Comment:
I added this log to be consistent with the log used in `makeFollowers()`. I think it may be useful for debugging purpose if anything goes wrong due to this change. I am not strong on this. I will remove it if you prefer.